### ISSUE TITLE
Checkboxes are not actionable using keyboard with NVDA

### ISSUE THREAD

**Issue by schne324** [dequelabs/cauldron#584]
<dl><dt>Description</dt><dd>Action cannot be performed with a screen reader turned on.
Unable to select checkbox using the keyboard. On pressing the space/enter key to check/uncheck the checkbox, the keyboard focus jumps to the next item. 

When the focus is present on the checkbox, it does not automatically change to forms mode, so the checkbox name, role, and the state are not announced to the screen reader users and the action is not performed.</dd><dt>Impact</dt><dd>Critical</dd><dt>Recommendation</dt><dd>RULE : 
Functionality MUST be available using screen reader touch methods (e.g. click actions), unless the functionality cannot be accomplished in any known way using a touch device.

HOW TO FIX: 
Fix this issue by ensuring that there are alternative methods to perform all gesture-based actions, for example: previous and next arrow buttons to advance a carousel in addition to swiping left and right.

REFERENCE: 
Deque University: https://dequeuniversity.com/class/input-methods2/touch-input/touch-with-screen-reader
Deque University: https://dequeuniversity.com/class/input-methods2/touch-input/functionality

BACKGROUND: 
When you turn on a screen reader on a touch device, the screen reader disables any scripted gestures on a web page because the screen reader itself relies on gestures to function. On iOS devices, for example, swiping to the right moves the screen reader focus to the next element, and swiping to the left moves the screen reader focus to the previous element. Swiping up and down on iOS devices invokes other screen reader behaviors. If a web page relies on swiping actions, those actions will not work when the screen reader is turned on.</dd><dt>Source code</dt><dd><pre><code><div aria-hidden="true" class="Icon Icon--checkbox-unchecked Checkbox__overlay"><svg xmlns="http://www.w3.org/2000/svg" overflow="visible" preserveAspectRatio="none" viewBox="1 1 22 22" height="24" width="24"><path d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z" vector-effect="non-scaling-stroke" fill="currentColor"></path></svg></div></code></pre></dd><dt>Checkpoint</dt><dd>Keyboard Navigation (2.1.1.a)</dd><dt>URL</dt><dd>https://cauldron.dequelabs.com/components/Checkbox</dd></dl>

**Comment by VaniChinnam** [comment]
Verified this ticket on axe pro - web page. The Keyboard user is able to select/ deselect the checkbox now with the NVDA ON.
